src/App.jsx

```diff
@@ -14,45 +21,78 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const updatedGoodsData = goodsFromServer
+  .map(good => ({ id: getRandomKey(), name: good, length: good.length }));
+
+export const App = () => {
+  const [sortField, setSortField] = useState(SORT_BY_NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(updatedGoodsData, {
+    sortField,
+    isReversed,
+  });
+
+  const resetGoods = () => {
+    setSortField(SORT_BY_NONE);
+    setIsReversed(false);
+  };
+
+  const isVisibleButton = sortField || isReversed;
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(SORT_BY_ALPHABHET)}
+          type="button"
+          className={
+            changeColorIfPressed(SORT_BY_ALPHABHET !== sortField, 'is-info')
+          }
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortField(SORT_BY_LENGTH)}
+          type="button"
+          className={
+            changeColorIfPressed(SORT_BY_LENGTH !== sortField, 'is-success')
+          }
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => setIsReversed(!isReversed)}
+          type="button"
+          className={
+            changeColorIfPressed(!isReversed, 'is-warning')
+          }
+        >
+          Reverse
+        </button>
```

Comment:
In future, when you gonna use typescript, don't forget that you can iterate through 'enum' of sort type options, and render these buttons in easier way